sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/graphql/generate/SchemaGenerator.java

```diff
@@ -4,17 +4,108 @@
+  List<GraphQLObjectType> objectTypes = new ArrayList<>();
+
   public GraphQLSchema generate(SqrlSchemaForInference schema) {
     SchemaGeneratorContext context = new SchemaGeneratorContext();
-    GraphQLSchema.Builder builder = GraphQLSchema.newSchema();
-    builder.query(schema.accept(new QueryTypeGenerator(), context));
-    builder.additionalTypes(schema.accept(new ObjectTypeGenerator(), context));
-    return builder.build();
+    schema.accept(new QueryTypeGenerator(queryFields), context);
+    schema.accept(new ObjectTypeGenerator(objectTypes), context);
+
+    postProcess();
+
+    if (queryFields.isEmpty()) {
+      throw new RuntimeException("No tables found to build schema");
+    }
+
+    GraphQLObjectType query = GraphQLObjectType.newObject()
+        .name("Query")
+        .fields(queryFields)
+        .build();
+
+    return GraphQLSchema.newSchema()
+        .query(query)
+        .additionalTypes(new LinkedHashSet<>(objectTypes))
+        .build();
+  }
+
+
+  void postProcess() {
+    // Ensure every field points to a valid type
+    Iterator<GraphQLObjectType> iterator = objectTypes.iterator();
+    while (iterator.hasNext()) {
+      GraphQLObjectType objectType = iterator.next();
+      List<GraphQLFieldDefinition> invalidFields = new ArrayList<>();
+
+      for (GraphQLFieldDefinition field : objectType.getFields()) {
+        if (!isValidType(field.getType())) {
+          invalidFields.add(field);
+        }
+      }
+
+      // Refactor to remove invalid fields
+      List<GraphQLFieldDefinition> fields = new ArrayList<>(objectType.getFields());
+      fields.removeAll(invalidFields);
+
+      // After removing invalid fields, if an object has no fields, it should be removed
+      if (fields.isEmpty()) {
+        iterator.remove();
+      }
+    }
+
+    queryFields.removeIf(field -> !isValidType(field.getType()));
+
+    // Ensure each object has at least one field
+    objectTypes.removeIf(objectType -> objectType.getFields().isEmpty());
+  }
+
+  boolean isValidType(GraphQLType type) {
+    type = unbox(type);
+    // You can expand this logic depending on the intricacies of type validation
+    if (type instanceof GraphQLTypeReference) {
+      GraphQLTypeReference typeReference = (GraphQLTypeReference)type;
+      for (GraphQLObjectType objectType : this.objectTypes) {
+        if (typeReference.getName().equalsIgnoreCase(objectType.getName())) {
+          return true;
+        }
+      }
+    }
+
+    return isBaseGraphQLType(type);
+  }
+
+  private GraphQLType unbox(GraphQLType type) {
+    if (type instanceof GraphQLNonNull) {
+      return unbox(((GraphQLNonNull) type).getWrappedType());
+    } else if (type instanceof GraphQLList) {
+      return unbox(((GraphQLList) type).getWrappedType());
+    }
+    return type;
+  }
+
+  public static boolean isValidGraphQLName(String name) {
+    return Pattern.matches("[_A-Za-z][_0-9A-Za-z]*", name);
```

Comment:
Is there no such method in the GraphQL library?

Reply:
No. It's a private variable. I copied it out of there.